src/Cache/CacheRegistry.php

```diff
@@ -79,11 +79,7 @@ protected function _create(object|string $class, string $alias, array $config):
         }
         unset($config['className']);
 
-        if (!($instance instanceof CacheEngine)) {
-            throw new CakeException(
-                'Cache engines must use Cake\Cache\CacheEngine as a base class.'
-            );
-        }
+        assert($instance instanceof CacheEngine, 'Cache engines must extend Cake\Cache\CacheEngine.');
```

Comment:
The `@throws` doesn't match anymore. There still is a `CakeException`, but for a different reason.